drivers/ethernet/Kconfig.xmc4xxx

```diff
@@ -46,6 +46,12 @@ config ETH_XMC4XXX_VLAN_HW_FILTER
 	  called multiple times, the filtering will be done on the latest
 	  tag.
 
+config ETH_XMC4XXX_MAC_EEPROM_ADDRESS
+	hex "EEPROM address of MAC"
+	default 0x0
+	help
+	  EEPROM address where the MAC address is stored.
```

Comment:
We shouldn't be mixing Kconfig with device tree properties, it should be possible to have this added in the device tree:

```
mac-eeprom = <&eeprom 0x00>;
```

Reply:
@pdgendt that would mean I'd have to modify the binding of the eeprom to add the -cells property?
@henrikbrixandersen would this be acceptable?

Reply:
If we are to do this (and not just rely on an application loading the board-specific MAC address from wherever and configuring the network interface), I'd prefer we align with how the Linux kernel devicetree bindings do this.

Something along the lines of:
- https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/nvmem/nvmem-consumer.yaml
- https://elixir.bootlin.com/linux/latest/source/arch/arm/boot/dts/aspeed/aspeed-bmc-asrock-e3c246d4i.dts (example use)

Reply:
> @pdgendt that would mean I'd have to modify the binding of the eeprom to add the -cells property?

I don't think it needs that see [the docs](https://docs.zephyrproject.org/latest/build/dts/phandles.html#zero-or-more-nodes-with-metadata-phandle-array-type)

Reply:
@pdgendt, unless I'm missing something, it is the binding of the referenced node that describes how the specifier entries are parsed. (i.e. all the gpio and dma examples).

https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/devicetree/gpio.h#L69
From this example it's the gpio binding that needs to define gpio-cells.

Reply:
Correct. Please see the section on specifier spaces further down that page.

I'd still prefer using the same specifier space as the Linux kernel nvmem consumer bindings.

Reply:
@henrikbrixandersen, I'll work on the nvmem consumer binding.

Reply:
@henrikbrixandersen this an example from linux that matches what I'm trying to do:
![image](https://github.com/user-attachments/assets/a1c20ad6-14c9-4451-96f4-79503cdce17c)

But I have a devicetree question,  what would be the default compatible of mac_address in this case? I tried the same approach in zephyr. It works, but it doesn't say which compatible was used for this node:

![image](https://github.com/user-attachments/assets/410e773f-75a3-4b61-a095-c5d2993e62c3)